packages/ts-plugin/src/astro2tsx.ts

```diff
@@ -0,0 +1,49 @@
+import type { FileMapping } from './source-mapper';
+import { readFileSync } from 'fs';
+
+const ASTRO_DEFINITION = readFileSync(require.resolve('../astro.d.ts'));
+
+// Note that this is a bit of a hack until the new compiler with proper
+// source map support.
+
+interface Astro2TSXOptions {
+  filename: string | undefined;
+  isTsFile: boolean;
+}
+
+interface Astro2TSXResult {
+  code: string;
+  map: {
+    mappings: FileMapping
+  }
+}
+
+export function astro2tsx(code: string, options: Astro2TSXOptions): Astro2TSXResult {
+  const compiled =  transformContent(code);
+
+  const result: Astro2TSXResult = {
+    code: compiled,
+    map: {
+      mappings: []
+    }
+  };
+  
+  return result;
+}
+
+// This is hacky but it works for now
+function addProps(content: string, dtsContent: string): string {
+  let defaultExportType = 'Record<string, any>';
+  if(/interface Props/.test(content)) {
+    defaultExportType = 'Props';
```

Comment:
blah, yea this is pretty gross. I imagine our new parser should eventually be able to give us this information, so that you could just do something like `const {ast} = awair parse(code)` and have the exported types included.

Reply:
agreed though this is fine for now, as a first pass

Reply:
another short-term option: https://stackoverflow.com/a/38086176/4682119

Reply:
The new compiler doesn't parse the component script section so it's not going to give this info, will still need to run it through the ts compiler API. I can probably add this now though, to cover cases where it's not an interface for example.

Reply:
Ok, brought in ts-morph and am checking for the Props export based on the AST now, works with interfaces or types.